File: sistema-examenes-backend/src/main/java/com/sistema/examenes/servicios/ExamenService.java
package com.sistema.examenes.servicios;

import java.util.Set;

import org.springframework.stereotype.Service;

import com.sistema.examenes.entidades.Examen;

@Service
public interface ExamenService {

	Examen agregarExamen (Examen examen);
	
	Examen actualizarExamen (Examen examen);
	
	Set<Examen> obtenerExamenes();
	
	Examen obtenerExamen (Long examenId);
	
	void eliminarExamen (Long examenId);
}

File: sistema-examenes-frontend/src/app/services/examen.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baserUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class ExamenService {

  constructor(private http:HttpClient) { }

  public listarExamenes(){
    return this.http.get(`${baserUrl}/examen/`);
  }

  public agregarExamen(examen:any){
    return this.http.post(`${baserUrl}/examen/`,examen);
  }

  public eliminarExamen(examenId:any){
    return this.http.delete(`${baserUrl}/examen/${examenId}`);
  }
  
  public obtenerExamen(examenId:any){
    return this.http.get(`${baserUrl}/examen/${examenId}`);
  }

  public actualizarExamen(examen:any){
    return this.http.put(`${baserUrl}/examen/`,examen);
  }
}
